### PR TITLE
nflgame.live run With Threading

### PR DESCRIPTION
**Pull request by timkutcher**
New to contributing to public projects like this, so forgive me if I'm not going about this normally.

__tl;dr:__ Made it so `nflgame.live run()` method can be run in a thread that can be shutdown gracefully, maintaining current API of that method.

---

I was using the `nflgame.live` module with the `run()` method to fetch some live game data within a web application but wound up having problems being able to shutdown gracefully. I'm still not _entirely_ sure what the full issue was on my end, but I wasn't able to shutdown the server without multiple interrupts to stop both the main run, and the `nflgame.live run()`. This ultimately lead to some goofy things on a heroku deployment and heroku couldn't restart the server normally.

I added a `LiveRunner` class to just be a separate thread that runs the `run()` method but changes/adds a condition in the loop to check for a [threading.Event](https://docs.python.org/3/library/threading.html#threading.Event). In order to not break the current API, as well as not just copy the code to that thread run method, I just have that thread's `run()` invoke the original `run()` method and added a `stop_event` parameter for it, so that it passes it's own `threading.Event()`. Then the shutdown method just has to set it.

With that change, `time.sleep()` changes to `threading.Event().wait()` if `stop_event` is not `None`.

This change allows for the live polling to run in a background thread that can be shutdown gracefully at any point (while still keeping the normal stop parameter to shutdown after a date).

On my end, everything works fine now and I'm using it like so:
```py
runner = nflgame.live.LiveRunner(callback=on_new_live_data, active_interval=180,
                                 inactive_interval=1800, stop=END_OF_POOL, daemon=True)
runner.start()
atexit.register(lambda: runner.shutdown())
```

Do note that I only use the `completed` part of the callback, but I don't imagine it would affect the others? Tricky to test stuff without live games going on...

Also, I haven't done this specifically, but I believe you should also be able to use it like this as well:

```py
import nflgame
stop_run = threading.Event()
nflgame.live.run(callback_func, stop_event=stop_run)
# do stuff...
stop_run.set()  # Now run stops
```

Could be useful for web apps or jobs to fetch between certain intervals (when we know games are occurring).

Updated comments to document changes.

Let me know if there are any questions. Great/helpful API!

### CODE REVIEW

**Comment by derek-adair** [comment]
Hey there, great idea. Any PRs going forward should be on the dev branch.  If you could re-submit it against that I will get it in there.  I need to add this somewhere in the README.md

**Also, I've changed how this script works as well as how the Game object fetches data.**
